smack-tcp/src/main/java/org/jivesoftware/smack/tcp/XmppTcpTransportModule.java

```diff
@@ -711,7 +711,7 @@ protected XmppTcpNioTransport getTransport() {
         return tcpNioTransport;
     }
 
-    static final class EstablishingTcpConnectionStateDescriptor extends StateDescriptor {
+    public static final class EstablishingTcpConnectionStateDescriptor extends StateDescriptor {
```

Comment:
Why does this become public? If possible, Smack internals should not be `public` and stay part of the internal API.